contracts/src/L1/gateways/L1GatewayRouter.sol

```diff
@@ -223,7 +254,13 @@ contract L1GatewayRouter is OwnableUpgradeable, IL1GatewayRouter {
         bytes memory _data,
         uint256 _gasLimit
     ) public payable override onlyNotInContext {
-        IL1ScrollMessenger(messenger).sendMessage{value: msg.value}(_to, _amount, _data, _gasLimit, _msgSender());
+        IL1ScrollMessenger(messenger).sendMessage{value: msg.value}(
+            _to,
+            _amount,
+            abi.encode(_msgSender(), _data),
```

Comment:
How will the receiver decode the data? shouldn't that be done by the bridge, so that the receiver is called with `_data` and not with `abi.encode(_msgSender(), _data)`?